File: Code 3 Callouts/Code 3 Callouts/Models/Ambient/IAmbientBase.vb
Imports Rage

Namespace Models.Ambient

    Public Interface IAmbientBase

        Function Start() As Boolean
        Sub Init()
        Function IsEventStarted() As Boolean
        Function GetRequiredPeds() As Boolean
        Function GetNearbyPeds() As Boolean
        Function SpawnRequiredPeds() As Boolean
        Function CheckPeds() As Boolean
        Function EndBasedOnDistance() As Boolean
        Sub CreateEntityBlips()
        Sub Process()
        Sub Dispatch911Call(ByVal pPosition As Vector3)
        Sub [End]()
        Sub Delete()

        Property Active() As Boolean
        ReadOnly Property PedsRequired() As Integer
        ReadOnly Property CanUseExistingPeds() As Boolean

    End Interface

End Namespace

File: Code 3 Callouts/Code 3 Callouts/Models/Ambient/IAmbientBase.cs
using Rage;

namespace Stealth.Plugins.Code3Callouts.Models.Ambient
{

	internal interface IAmbientBase
	{

		bool Start();
		void Init();
		bool IsEventStarted();
		bool GetRequiredPeds();
		bool GetNearbyPeds();
		bool SpawnRequiredPeds();
		bool CheckPeds();
		bool EndBasedOnDistance();
		void CreateEntityBlips();
		void Process();
		void Dispatch911Call(Vector3 pPosition);
		void End();

		void Delete();
		bool Active { get; set; }
		int PedsRequired { get; }

		bool CanUseExistingPeds { get; }
	}

}